internal/maintenance/startstop/start_stop.go

```diff
@@ -97,19 +103,37 @@ type BaseStartStop struct {
 //	}
 //
 //	...
-func (s *BaseStartStop) StartInit(ctx context.Context) (context.Context, bool, chan struct{}) {
+func (s *BaseStartStop) StartInit(ctx context.Context) (context.Context, bool, func(), func()) {
```

Comment:
This is where I start disliking the lint rule discouraging named return values. I mainly just want to see a label on things when there are this many, especially when there are consecutive values of the same type.

Reply:
Yeah, definitely fair. I tried to write some good docs to compensate for it in this case. Will hopefully help.